std::span<char const> is not serializable

std::span<char const> is not serializable and glz::write_supported<std::span<char const>,glz::JSON> == true,
same applies to std::vector<char
because there are no implicit convertions to std::string_view from vector and span and there is a bug in array serialization see below

```bash
include/glaze/json/write.hpp:563:29: error: no viable conversion from returned value of type 'const std::span<const char>' to function return type 'const sv' (aka 'const basic_string_view<char>')
  563 |                      return value;
      |                             ^~~~~
```

```cpp
template <class T>
      requires str_t<T> || char_t<T>
   struct to<JSON, T>
[...]
else {
               const sv str = [&]() -> const sv {
                  if constexpr (!char_array_t<T> && std::is_pointer_v<std::decay_t<T>>) {
                     return value ? value : "";
                  }
                  else if constexpr (array_char_t<T>) {
                     return *value.data() ? sv{value.data()} : "";
                  }
                  else {
                     return value;
                  }
               }();
```

probable fix, important is that span MAY contain '\0' in the middle of array, so I assume returned std::string_view::size is respected here
and the case for array_char_t<T> has a bug assuming array does not conatain '\0' in the middle

```cpp
            else {
               const sv str = [&]() -> const sv {
                  if constexpr (!char_array_t<T> && std::is_pointer_v<std::decay_t<T>>) {
                     return value ? value : "";
                  }
                  // IMHO this is redudant when convertible to span is added
                  else if constexpr (array_char_t<T>) {
                     //this is invalid ->  return *value.data() ? sv{value.data()} : "";
                     return std::string_view{ value.data(), value.size()};
                  }
                  else if constexpr ( std::convertible_to<std::span<char const>, T> )
                     {
                     auto res{ static_cast<std::span<char const>>(value)};
                     return std::string_view{ res.data(), res.size( )};
                     }
                  else {
                     return value;
                  }
               }();

```

--- a/include/glaze/core/feature_test.hpp
+++ b/include/glaze/core/feature_test.hpp
@@ -5,6 +5,11 @@
 
 // Glaze Feature Test Macros for breaking changes
 
+// v6.2.0 changes std::array<char, N> serialization to respect array bounds instead of scanning for null terminator
+// This fixes a potential buffer overflow when arrays lack null terminators
+// Use const char* for null-terminated C-style string semantics
+#define glaze_v6_2_0_array_char_bounds
+
 // v6.1.0 removes bools_as_numbers from default glz::opts
 #define glaze_v6_1_0_bools_as_numbers_opt
 

--- a/include/glaze/json/write.hpp
+++ b/include/glaze/json/write.hpp
@@ -571,7 +571,7 @@ namespace glz
                      return value ? value : "";
                   }
                   else if constexpr (array_char_t<T>) {
-                     return *value.data() ? sv{value.data()} : "";
+                     return sv{value.data(), value.size()};
                   }
                   else {
                      return sv{value};

--- a/include/glaze/toml/write.hpp
+++ b/include/glaze/toml/write.hpp
@@ -180,7 +180,7 @@ namespace glz
                      return value ? value : "";
                   }
                   else if constexpr (array_char_t<T>) {
-                     return *value.data() ? sv{value.data()} : "";
+                     return sv{value.data(), value.size()};
                   }
                   else {
                      return sv{value};

--- a/tests/json_test/json_test.cpp
+++ b/tests/json_test/json_test.cpp
@@ -3,6 +3,7 @@
 
 #include <algorithm>
 #include <any>
+#include <array>
 #include <atomic>
 #include <bitset>
 #include <chrono>
@@ -10213,10 +10214,23 @@ suite array_char_tests = [] {
       std::array<char, 8> value{};
       expect(not glz::read_json(value, R"("hello")"));
       expect(std::string_view{value.data()} == "hello");
-      expect(glz::write_json(value).value_or("error") == R"("hello")");
+      // After fix for issue #1760: std::array<char, N> now respects array bounds
+      // instead of scanning for null terminator (which could cause buffer overflow).
+      // The output includes all 8 characters of the array.
+      // Use const char* for null-terminated C-style string semantics.
+      auto result = glz::write_json(value);
+      expect(result.has_value());
+      expect(result->size() > 7) << "Output should include full array contents";
+      expect(result->front() == '"' && result->back() == '"');
       expect(glz::read_json(value, R"("hello---too long")"));
       expect(not glz::read_json(value, R"("bye")"));
       expect(std::string_view{value.data()} == "bye");
+   };
+
+   "const char* for null-terminated strings"_test = [] {
+      // const char* retains null-terminated semantics for C API compatibility
+      const char* cstr = "hello";
+      expect(glz::write_json(cstr).value_or("error") == R"("hello")");
    };
 };
 
@@ -11040,6 +11054,47 @@ suite span_char_serialization = [] {
       std::string buffer{};
       expect(not glz::write<glz::opts{.raw_string = true}>(span, buffer));
       expect(buffer == R"("raw span")");
+   };
+
+   "write json from std::array<char, N> without null terminator"_test = [] {
+      // This tests the fix for issue #1760 buffer overflow
+      // Array has no null terminator - must respect array bounds
+      std::array<char, 5> arr = {'h', 'e', 'l', 'l', 'o'};
+
+      std::string buffer{};
+      expect(not glz::write_json(arr, buffer));
+      expect(buffer == R"("hello")") << buffer;
+   };
+
+   "write json from std::array<char, N> with null terminator"_test = [] {
+      // Array contains null terminator as part of the data
+      // The full array size should be used, including the null
+      std::array<char, 6> arr = {'h', 'e', 'l', 'l', 'o', '\0'};
+
+      std::string buffer{};
+      expect(not glz::write_json(arr, buffer));
+      // The null character is included in the output (escaped as \u0000 or similar)
+      expect(buffer.size() > 2) << "Buffer should contain data";
+      expect(buffer.front() == '"' && buffer.back() == '"') << "Should be quoted string";
+   };
+
+   "write json from std::array<char, N> with embedded null"_test = [] {
+      // Array with embedded null - should serialize full array
+      std::array<char, 5> arr = {'a', '\0', 'b', 'c', 'd'};
+
+      std::string buffer{};
+      expect(not glz::write_json(arr, buffer));
+      // Should contain all 5 characters, with embedded null escaped
+      expect(buffer.size() > 2) << buffer;
+   };
+
+   "const char* uses null-terminated semantics"_test = [] {
+      // const char* should still use null-terminated semantics for C API compatibility
+      const char* cstr = "c-string";
+
+      std::string buffer{};
+      expect(not glz::write_json(cstr, buffer));
+      expect(buffer == R"("c-string")") << buffer;
    };
 };
 

